### PR TITLE
OOP school library: unit tests

### PR DESCRIPTION
**Pull request by nedjwab**
### In This pull request we : 
- Add unit tests to your classes:
- Book
- Person
- Student
- Teacher
- Classroom
- Rental
- all classes that you created while implementing the Decorator design pattern

### CODE REVIEW

File: spec/base_spec.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    @base.should be_an_instance_of Base
+  end
+  it 'returns name is nedjwa' do
+    @base.nameable.should eql 'Nedjwa'
+  end
+end
+
+describe CapitalizeDecorator do
+  before :each do
+    @capitalize = CapitalizeDecorator.new('Nedjwa')
```
Comment (FrederickMih) [inline spec/base_spec.rb]: - [ ] Kindly note that this method capitalizes the first letter of a word. If I change the name to `nedjwa`, this test should still pass which is not the case right now. Kindly fix this issue.
- **Hint** I think there's a difference between `upcase()` and `capitalize()` methods in Ruby

File: spec/person_spec.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+require_relative '../person'
+require 'yaml'
+
+describe Person do
+  before :each do
+    @person = Person.new(25, 'bello', 'teacher')
+  end
+  it 'returns persons name bello' do
+    @person.correct_name.should eql 'bello'
```
Comment (FrederickMih) [inline spec/person_spec.rb:9]: - [**Optional**] The syntax `should` is deprecated(a warning from the console) hence let's use the new `expect` syntax
- **Hint**: This line can be rewritten as `expect(@person.age).to eql 25`

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



# To Highlights 

- Github workflow has been respected :heavy_check_mark: 
- PR has a good title and summary :heavy_check_mark: 
- No Linter errors :+1: 
- Descriptive README :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sja-thedude — APPROVED** [review]
## Project Approved 👍 

Hi @iBel251, This is @sja-thedude your code reviewer.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

giphy

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**